sound/hda/hdac_stream.c

```diff
@@ -405,14 +406,20 @@ int snd_hdac_stream_setup_periods(struct hdac_stream *azx_dev)
 {
 	struct hdac_bus *bus = azx_dev->bus;
 	struct snd_pcm_substream *substream = azx_dev->substream;
+	struct snd_compr_stream *cstream = azx_dev->cstream;
 	struct snd_pcm_runtime *runtime;
 	struct snd_dma_buffer *dmab;
 	__le32 *bdl;
 	int i, ofs, periods, period_bytes;
 	int pos_adj, pos_align;
 
-	runtime = substream->runtime;
-	dmab = snd_pcm_get_dma_buf(substream);
+	if (substream) {
+		runtime = substream->runtime;
+		dmab = snd_pcm_get_dma_buf(substream);
+	} else {
```

Comment:
@crojewsk dont you need else if (cstream) here?

Reply:
_setup_periods can only be called during hda stream setup during preparation for PCM -or- (now) Compress runtime. As you can see, there is no NULL-check for PCM-only case as this would be redundant. Adding cstream variant does not change that. Superfluous ifs only complicate the code, unnecessarily.

Reply:
@crojewsk didnt quite get this. You have if (substream) on line 416, so I was wondering if you also need a if (cstream) on line 419?

Reply:
Believe I answered that. TLDR:
current code: no checks, PCM != NULL is implicit paradigm
after compr addition: no checks, PCM != NULL || compr != NULL is implicit paradigm
No redundancy.

The only reason substream check is there is for correct locals initialization.

Reply:
OK got it. thanks!